source/game_sa/Scripts/CommandParser/ReadArg.hpp

```diff
@@ -116,57 +109,36 @@ inline T Read(CRunningScript* S) {
     } else if constexpr (std::is_same_v<Y, CRect>) {
         return { Read<CVector2D>(S), Read<CVector2D>(S) };
     } else if constexpr (std::is_same_v<Y, std::string_view>) { 
-        auto& IP = S->m_IP;
-
-        const auto FromScriptSpace = [](const auto offset) {
-            return (const char*)&CTheScripts::ScriptSpace[offset];
-        };
-
-        const auto FromGlobalArray = [&](uint8 elemsz) {
-            const auto [idx, offset] = detail::ReadArrayInfo(S);
-            return FromScriptSpace(elemsz * idx + offset);
-        };
-
-        const auto FromLocalArray = [&](uint8 elemsz) {
-            const auto [idx, offset] = detail::ReadArrayInfo(S);
-            return (const char*)S->GetPointerToLocalArrayElement(offset, idx, elemsz);
-        };
-
-        const auto FromStaticString = [&](size_t strsz) -> std::string_view {
-            const auto str = (const char*)(IP);
-            IP += strsz;
-            return str;
-        };
-
-        switch (const auto ptype = (eScriptParameterType)S->ReadAtIPAs<int8>()) {
+        switch (const auto ptype = (eScriptParameterType)S->GetAtIPAs<int8>()) {
         case SCRIPT_PARAM_GLOBAL_SHORT_STRING_VARIABLE:
-            return FromScriptSpace(S->ReadAtIPAs<int16>());
+            return S->GetGlobal<scm::ShortString>(S->GetAtIPAs<int16>());
```

Comment:
```suggestion
            return S->GetGlobal<scm::ShortString>(S->GetAtIPAs<uint16>());
```
This needs to unsigned (bug in current implementation)